troubleshooting-guide/widget-errors.md

```diff
@@ -167,3 +167,32 @@ Table1 -> Table1.tableData
 Table1.isVisible -> Table1
 ```
 
+#Debugging
+
+For debugging JS logic inside widget actions, you can use `debugger` statement.
+
+##Description
+
+The debugger statement invokes any available debugging functionality, such as setting a breakpoint. If no debugging functionality is available, this statement has no effect.
+
+##Syntax
+
+```text
+debugger;
+```
+
+##Examples
+
+###Using the debugger statement
+
+The following example shows code where a debugger statement has been inserted, to invoke a debugger (if one exists) when the function is called.
+
+```text
+function potentiallyBuggyCode() {
+debugger;
+// do potentially buggy stuff to examine, step through, etc.
+}
+```
+
+When the debugger is invoked, execution is paused at the debugger statement. It is like a breakpoint in the script source.
```

Comment:
We should mention that one needs to open the browser console to make it pause execution

Reply:
Done